src/ManifestStrategy/WireNavigationManifestStrategy.php

```diff
@@ -0,0 +1,208 @@
+                'The project doesn\'n have the navigation source file: %s.',
+                static::TARGET_NAVIGATION_FILE,
+            ), InputOutputInterface::DEBUG);
+
+            return false;
+        }
+
+        try {
+            $navigationConfiguration = $this->getNavigationConfiguration();
+        } catch (UnexpectedNavigationXmlStructureException $unexpectedNavigationXmlStructureException) {
+            return false;
+        }
+        
+        $navigationConfiguration = $this->applyNewNavigationConfigurations(
+            $navigationConfiguration,
+            $manifest[static::KEY_NAVIGATIONS_CONFIGURATION],
+            $manifest[static::KEY_NAVIGATION_POSITION_BEFORE] ?? null,
+            $manifest[static::KEY_NAVIGATION_POSITION_AFTER] ?? null,
+        );
+        $this->saveNavigationConfigurationsIntoSourceFile($navigationConfiguration);
+
+        return true;
+    }
+
+    /**
+     * @return array<string|int, array|string>
+     */
+    protected function getNavigationConfiguration(): array
+    {
+        $mainNavigationXmlElement = simplexml_load_file($this->getNavigationSourceFilePath());
+        $mainNavigationAsJson = json_encode($mainNavigationXmlElement);
+
+        if ($mainNavigationAsJson === false) {
+            throw new UnexpectedNavigationXmlStructureException();
+        }
+
+        return json_decode($mainNavigationAsJson, true);
+    }
+
+    /**
+     * @param array<string|int, array|string> $navigationConfiguration
+     * @param array<string|int, array|string> $newNavigationConfiguration
+     * @param string|null $before
+     * @param string|null $after
+     *
+     * @return array<string|int, array|string>
+     */
+    protected function applyNewNavigationConfigurations(
+        array $navigationConfiguration,
+        array $newNavigationConfiguration,
+        ?string $before = null,
+        ?string $after = null
+    ): array {
+        $resultNavigationConfiguration = [];
+        $itemAdded = false;
+
+        foreach ($navigationConfiguration as $navigationItemKey => $navigationItemConfiguration) {
+            if ($navigationItemKey === $before) {
+                $this->injectNewNavigationConfigurations($resultNavigationConfiguration, $navigationConfiguration, $newNavigationConfiguration);
+                $resultNavigationConfiguration[$navigationItemKey] = $navigationItemConfiguration;
+                $itemAdded = true;
+
+                continue;
+            }
+
+            if ($navigationItemKey === $after) {
+                $resultNavigationConfiguration[$navigationItemKey] = $navigationItemConfiguration;
+                $this->injectNewNavigationConfigurations($resultNavigationConfiguration, $navigationConfiguration, $newNavigationConfiguration);
+                $itemAdded = true;
+
+                continue;
+            }
+
+            $resultNavigationConfiguration[$navigationItemKey] = $navigationItemConfiguration;
+        }
+
+        if (!$itemAdded) {
```

Comment:
@matweew Here it would be added to the end of the navigation when `before` or `after` was not found.

Wouldn't something like this work as well?

```
$key = $before ?? $after;
$position = array_search($key, array_keys($navigationConfiguration));

if (!$position) { // Neither before or after was found in the given navigation, add the element to the end.
    $this->injectNewNavigationConfigurations($resultNavigationConfiguration, $navigationConfiguration, $newNavigationConfiguration);

    return $navigationConfiguration;
}

if ($before !== null) {
   $position--; 
} 

return array_slice($navigationConfiguration, 0, $position, true) 
    + $newNavigationConfiguration
    + array_slice($navigationConfiguration, $position, count($navigationConfiguration) - 1, true) ;
```

Reply:
yes, adding to the end of the list is OK to me.